Spec.md

```diff
@@ -0,0 +1,227 @@
+- State
+
+The properties Name, NumberInList, Id, and Number are fields of the DataSets. The value of CurrentState is mapped as a field of State and has a DataSet with its properties:
+
+- Production
+- ActiveProgram
+  - Name
+  - NumberInList
+  - CurrentState
+- State
+- CurrentState
+  - Id
+  - Number
+
+### Explanation
+
+A field can also contain properties but no other variables, even though the address space can contain more hierarchies. Properties of objects also cannot be mapped to the properties of a field. Therefore, a uniform mapping is used for objects and variables so that access is always consistent.
+
+## Mapping of References
+
+The name of a DataSet is created by the BrowsePath of the object. The NamespaceUri is used instead of the NamespaceIndex to make the path server-independent. Other BrowsePaths may also be possible. Alternative BrowsePaths can be mapped to the description (see description field of the MetaData). References can also be mapped into the description.
+
+## Topic Structure
+
+The generic topic structure for OPC UA is:
+
+`<Prefix>/<Encoding>/<MqttMessageType>/<PublisherId>/[<WriterGroup>[/<DataSetWriter>]]`
+
+For the umati Dashboard, this topic structure is restricted as follows:
+
+- `<Prefix> = umati.v3`
+- `<Encoding> = json`
+- `<MqttMessageType> = as defined in OPC UA PubSub`
+- `<PublisherId> = Name of the Client`
+- `<WriterGroup> = Expanded NodeId of the object representing the machine`
+- `<DataSetWriter> = BrowsePath`
+
+The BrowsePath in the DataSetWriter is built with a "." between each BrowseName. The BrowseNames use only the namespace index to avoid collisions.
+Special character as `'/''.''<'| '>' | ':' | '#' | '!' | '&' ` in the `name` field of the BrowseName  need to encode by [URL-Encoding](https://de.wikipedia.org/wiki/URL-Encoding) using an underscore instead of a '%'.
+For identification of the object, the name of the DataSet with expanded NodeId must be used.
+If two node has the same BrowsePath a iterator ("_Number") can be send to avoid collisions (e.g, `3:Partent.3:Tool_1`, `3:Partent.3:Tool_2` `3:Partent.3:Tool_3` )
+
+### Examples
+
+DataSet Topic
+
+```text
+umati.v3/json/data/example_publisher_1/nsu=http:_2F_2Fexample.com_2FShowcaseMachineTool_2F;i=66382/2:Identification
+umati.v3/json/data/example_publisher_1/nsu=http:_2F_2Fexample.com_2FShowcaseMachineTool_2F;i=66382/3:Production.3:ActiveProgram
+umati.v3/json/data/example_publisher_1/nsu=http:_2F_2Fexample.com_2FShowcaseMachineTool_2F;i=66382/3:Production.3:ActiveProgram.1:State
+```
+
+MetaDataSet Topic
+
+```text
+umati.v3/json/metadata/example_publisher_1/nsu=http:_2F_2Fexample.com_2FShowcaseMachineTool_2F;i=66382/2:Identification
+umati.v3/json/metadata/example_publisher_1/nsu=http:_2F_2Fexample.com_2FShowcaseMachineTool_2F;i=66382/3:Production.3:ActiveProgram
+umati.v3/json/metadata/example_publisher_1/nsu=http:_2F_2Fexample.com_2FShowcaseMachineTool_2F;i=66382/3:Production.3:ActiveProgram.1:State
+```
+
+### Alternative

```

Comment:
@ccvca @wlkrm What is your opinion about the different topic mappings?

Reply:
@GoetzGoerisch I like your suggestion and for me it is okay that we do not follow the recommendation of Part 14. 
Can we generalies your suggestion, that the conecpt works in other context?
Changes to your suggestion: 
<InstancesExpandedNodeId> -> also the BrowseName of the Node (with a collision protection see above)
<TypeDefinitionEntryNode> -> optional an before the machines than the topic looks like:

```text
umati/v3/opcua/json/data/<PublisherId>/<TypeDefinitionEntryNode>/<BrowsePath>/
```

e.g.,

```text
umati/v3/opcua/json/data/<PublisherId>/MachineToolType/machines/FullMachineExample/MachineryBuildingBlocks/MachineryItemState
umati/v3/opcua/json/data/<PublisherId>/MachineToolType/machines/FullMachineExample_1/MachineryBuildingBlocks/MachineryItemState  
```

A question of this how to define which TypeDefinitionEntryNode? It is custombased which type he want to use?